### PR TITLE
[9648] Pretty PyPI page

### PR DESCRIPTION
**Pull request by twm**
## Summary

* Adjust URLs to HTTPS instead of HTTP
* Add project URLs for documentation, source, and issues. They will display on PyPI like this: 
screenshot

* Generate the long description of the package from README.rst. A slightly hackish regex replacement is used to make relative URLs absolute so that they will work when it is displayed at https://pypi.org/project/Twisted/ . Refer to https://packaging.python.org/specifications/core-metadata/#description-content-type.
* Add a Tox environment that runs the `twine check` command. This validates that the reStructuredText of the long description is valid. (I explored adding a test in `test_setup.py` for this, but [readme_renderer](https://pypi.org/project/readme_renderer/) isn't really documented for API use and has a pretty hefty dependency tree, so I found it a bit much for the assurance it would offer.)
* Remove some badges that were rendered obsolete by the switch to Coveralls and Azure Piplelines.

## Contributor Checklist:

* [x] There is an associated ticket in Trac. https://twistedmatrix.com/trac/ticket/9648#comment:1
* [x] The changes pass minimal style checks (see: https://twistedmatrix.com/trac/wiki/TwistedDevelopment#GettingYourPatchAccepted )
* [x] I have created a newsfragment in src/twisted/newsfragments/ (see: https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles )
* [x] I have updated the automated tests.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -41,10 +39,10 @@ Additional instructions for installing this software are in `the installation in
 Documentation and Support
 -------------------------
 
-Twisted's documentation is available from the `Twisted Matrix website <http://twistedmatrix.com/documents/current/>`_.
+Twisted's documentation is available from the `Twisted Matrix website <https://twistedmatrix.com/documents/current/>`_.
```
Comment (glyph) [inline README.rst:42]: Thanks for fixing this.

File: README.rst [hunk 2]
```diff
@@ -2,9 +2,7 @@ Twisted
 =======
 
 |pypi|_
-|coverage|_
```
Comment (glyph) [inline README.rst:5]: Not that I disagree, but why did you remove these, out of curiosity?

File: src/twisted/python/test/test_setup.py [hunk 3]
```diff
@@ -69,7 +69,7 @@ def test_win32Definition(self):
 
 
 
-class OptionalDependenciesTests(TestCase):
+class OptionalDependenciesTests(SynchronousTestCase):
```
Comment (glyph) [inline src/twisted/python/test/test_setup.py:72]: ❤️

File: src/twisted/python/test/test_setup.py [hunk 4]
```diff
@@ -363,3 +363,39 @@ def test_find_package_modules(self):
             ]),
             sorted(result),
         )
+
+
+
+class LongDescriptionTests(SynchronousTestCase):
+    """
+    Tests for C{_getLongDescriptionArgs()}
```
Comment (glyph) [inline src/twisted/python/test/test_setup.py:371]: Nitpick: direct references like this are generally an `L{}`.

**Review by glyph — APPROVED** [review]
I love everything about this.

File: README.rst (hunk 2 above)
Comment (twm) [reply README.rst:5]: This is the codecov badge. We switched to Coveralls, no?

File: src/twisted/python/test/test_setup.py (hunk 4 above)
Comment (twm) [reply src/twisted/python/test/test_setup.py:371]: :sob:

I seem to be unable to internalize epytext.

**Comment by twm** [comment]
Thanks for the review, @glyph!

File: README.rst (hunk 2 above)
Comment (glyph) [reply README.rst:5]: We did, but they have a badge too :)

Comment (glyph) [reply README.rst:5]: Oh, and I guess we're going to turn off Appveyor now that we have Azure?
